app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -17,12 +17,14 @@ import org.mozilla.fenix.home.recentbookmarks.RecentBookmark
 import org.mozilla.fenix.home.recentsyncedtabs.RecentSyncedTabState
 import org.mozilla.fenix.home.recenttabs.RecentTab
 import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem
+import org.mozilla.fenix.library.history.PendingDeletionHistory
 import org.mozilla.fenix.gleanplumb.Message
 import org.mozilla.fenix.gleanplumb.MessagingState
 
 /**
  * [Action] implementation related to [AppStore].
  */
+@Suppress("UndocumentedPublicClass")
```

Comment:
nit: the convention here is to add new classes to `detekt-baseline.xml`

Reply:
I am a bit confused, what should I write about?
```
data class AddPendingDeletionSet(val historyItems: Set<PendingDeletionHistory>) : AppAction()
    data class UndoPendingDeletionSet(val historyItems: Set<PendingDeletionHistory>) : AppAction()
```
The name of the class tells what it does already. `pendingDeletionHistoryItems` are already documented inside the `AppState`.

This new lint rule is a bit too demanding.

Reply:
added documentation